v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java

```diff
@@ -189,6 +189,54 @@ public void migrationTestWithInsertsOnly() {
     assertMovieTableContents();
   }
 
+  @Test
+  public void interleavedAndFKAndIndexTest() {
+    ChainedConditionCheck conditionCheck =
+        ChainedConditionCheck.builder(
+                List.of(
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Articles",
+                        "mysql-Articles.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Articles.jsonl"),
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Authors",
+                        "mysql-Authors.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Authors.jsonl"),
+                    uploadDataStreamFile(
+                        jobInfo,
+                        "Books",
+                        "mysql-Books.jsonl",
+                        "DataStreamToSpannerSimpleIT/mysql-Books.jsonl"),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Articles")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build(),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Books")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build(),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Authors")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build()))
+            .build();
+
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(12)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    // Assert specific rows
+    assertAuthorsTable();
```

Comment:
To validate that indexes are created and have data will it be better to query the indexes like below?
`SELECT AlbumId, AlbumTitle, MarketingBudget
    FROM Albums@{FORCE_INDEX=AlbumsByAlbumTitle}`

Reply:
Done.